src/prognostic_equations/hyperdiffusion.jl

```diff
@@ -40,7 +40,7 @@ function hyperdiffusion_cache(Y, hyperdiff::ClimaHyperdiffusion, turbconv_model)
             ᶜ∇²mseʲs = similar(Y.c, NTuple{n, FT}),
             ᶜ∇²q_totʲs = similar(Y.c, NTuple{n, FT}),
         ) :
-        turbconv_model isa DiagnosticEDMFX ? (; ᶜ∇²tke⁰ = similar(Y.c, FT)) :
+        use_prognostic_tke(turbconv_model) ? (; ᶜ∇²tke⁰ = similar(Y.c, FT)) :
```

Comment:
It should be the same for PrognosticEDMFX, where we only add ᶜ∇²tke⁰ to the cache if `use_prognostic_tke` (all other variables e.g. ᶜ∇²uₕʲs should always be in the cache though). Could you change that?

Reply:
Changed